src/format/fmt.go

```diff
@@ -0,0 +1,78 @@
+// Package format does autoformatting of BUILD files.
+//
+// It is based on a mildly modified version of buildifier; that supports fstrings
+// but not some of the other dialetical differences (e.g. type annotations).
+package format
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+
+	"github.com/bazelbuild/buildtools/build"
+	"gopkg.in/op/go-logging.v1"
+
+	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/fs"
+	"github.com/thought-machine/please/src/utils"
+)
+
+var log = logging.MustGetLogger("format")
+
+// Format reformats the given BUILD files to their canonical version.
+// It either prints the reformatted versions to stdout or rewrites the files in-place.
+// If no files are given then all BUILD files under the repo root are discovered.
+// The returned bool is true if any changes were needed.
+func Format(config *core.Configuration, filenames []string, rewrite, quiet bool) (bool, error) {
+	if len(filenames) == 0 {
+		return formatAll(utils.FindAllBuildFiles(config, core.RepoRoot, ""), rewrite, quiet)
+	}
+	ch := make(chan string)
+	go func() {
+		for _, filename := range filenames {
+			ch <- filename
+		}
+		close(ch)
+	}()
+	return formatAll(ch, rewrite, quiet)
```

Comment:
Why did you fan out here? Unless I'm mistaken there's still only one thread doing the formatting. Did you want to spin up some go routines?

Reply:
it's not fanning out, it's just changing from a slice to a channel to match what FindAllBuildFiles does